R/MoBiProject.R

```diff
@@ -0,0 +1,170 @@
+    getObservedData = function(dataSetNames = NULL){
+    },
+
+    #' @description
+    #' Get modules present in the project.
+    #'
+    #' @returns A named list of `MoBiModule` objects. `NULL` if the project does not contain
+    #' any module.
+    getModules = function(){
+    },
+
+    #' @description
+    #' Get a specified individual from the project
+    #' @param name Name of the individual
+    #' @param stopIfNotFound If `TRUE` (default), an error is thrown if the specified
+    #' individual is not present in the project.
+    #'
+    #' @returns An object of the type `BuildingBlock`. `NULL` if the project does not contain
+    #' such an individual and `stopIfNotFound = FALSE`.
+    getIndividual = function(name, stopIfNotFound = TRUE){
+    },
+    #' @description
+    #' Get specified expression profiles from the project.
+    #' @param names List of names of the expression profiles to retrieve.
+    #' @param stopIfNotFound If `TRUE` (default), an error is thrown if any of the specified
+    #' expression profiles is not present in the project.
+    #' @returns A named list of objects of the type `BuildingBlock`. `NULL` for each
+    #' specified expression profile that is not present in the project  if `stopIfNotFound = FALSE`.
+    getExpressionProfiles = function(names, stopIfNotFound = TRUE){
+    },
+
+    #' @description
+    #' Create a simulation configuration.
+    #'
+    #'
+    #' @param modulesNames A list of the modules from which to create in simulation.
+    #' All defined modules must be present in the project. The order of module names defines the order in which the modules will be combined to a simulation!
+    #' @param individualName Optional, name of the individual.
+    #' @param expressionProfilesNames Optional, list of expression profiles to apply to the simulation.
+    #' @param selectedInitialConditions By default, the first Initial Conditions
+    #' (IC) building block (BB) of each module will be selected. If a module has multiple
+    #' IC BBs, it is possible to specify which IC BB to apply by providing a named list,
+    #' where the name should be the name of the module and the value the name of the IC BB.
+    #' By setting the value to `NULL`, no IC BB from the specified module will be applied.
+    #' @param selectedParameterValues By default, the first Parameter Values
+    #' (PV) building block (BB) of each module will be selected. If a module has multiple
+    #' PV BBs, it is possible to specify which PV BB to apply by providing a named list,
+    #' where the name should be the name of the module and the value the name of the PV BB.
+    #' By setting the value to `NULL`, no PV BB from the specified module will be applied.
+    #'
+    #' @returns A `SimulationConfiguration` object.
+     createSimulationConfiguration = function(modulesNames, individualName = NULL, expressionProfilesNames = NULL, selectedInitialConditions = NULL, selectedParameterValues = NULL){
```

Comment:
So, is the plan to have the moduleNames order and selected lists order related? I would say the better way to handle this is that we mirror the .NET construction. You need a set of `ModuleConfigurations` to create a `SimulationConfiguration` The `ModuleConfiguration` indicates which, if any, PV and IC are selected.